### PR TITLE
Update release instructions

### PR DESCRIPTION
**Pull request by ianlewis**
Updates the release instructions to create a pre-release tag at the beginning so we can update tag references before pre-release testing.

Signed-off-by: Ian Lewis <ianlewis@google.com>

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]
Thanks!

**Comment by laurentsimon** [comment]
Do you want to temporarily add a command for documentation, before https://github.com/slsa-framework/slsa-github-generator/pull/1193 is merged?

File: RELEASE.md [hunk 1]
```diff
@@ -31,20 +32,37 @@ $ export BUILDER_REPOSITORY="$GITHUB_USERNAME/slsa-github-generator"
...
+Click `Publish release`.
+
+## Verify version references & code freeze
 
-Verify the references to the internal Actions by manually running the [release workflow](https://github.com/slsa-framework/slsa-github-generator/actions/workflows/release.yml). Ensure this workflow succeeds. If the references are still `@main`, update them with the following command:
+Update version references with the following command:
 
 ```shell
-find .github/workflows/ -name '*.yaml' -o -name '*.yml' | xargs sed -i 's/uses: slsa-framework\/slsa-github-generator\/\.github\/actions\/\(.*\)@main*/uses: slsa-framework\/slsa-github-generator\/.github\/actions\/\1@_YOUR_RELEASE_TAG_/'
+find .github/workflows/ -name '*.yaml' -o -name '*.yml' | xargs sed -i "s/uses: slsa-framework\/slsa-github-generator\/\.github\/actions\/\(.*\)@main*/uses: slsa-framework\/slsa-github-generator\/.github\/actions\/\1@$BUILDER_TAG/"
```
Comment (pnacht) [inline RELEASE.md:51]: In other parts of the documentation, placeholder "tags" have been displayed as `<BUILDER_TAG>` to clearly differentiate them from real tags or environment variables.

Comment (ianlewis) [reply RELEASE.md:51]: This is a real environment variable that we set in the beginning.

**Comment by ianlewis** [comment]
> Do you want to temporarily add a command for updating the documentation, before #1193 is merged?

There is a command that was added in a previous PR. Though I think it could be more generic. I updated the command to update versions in documentation so that it works when the previous version is 'main' or any other semantic version.

**Pull request merged**